reactioncommerce/reaction
# feat: add `removeAccountGroup` mutation, update `updateAccountGroup` mutation

Resolves #5946  
Impact: **minor**  
Type: **feature**

## Issue

- Missing `removeAccountGroup` mutation and integration tests

## Solution

- Add missing `removeAccountGroup` mutation and integration tests
- Update `updateAccountGroup` mutation
  - It no longer updates user roles when the group is updated

## Breaking changes

none

## Testing
1. Create an account
1. Create a new group
1. Add that account to the new group
1. Make sure they're in the new group
1. Remove that account from the new group
1. Verify the account has been moved the the `customer` group
1. Attempt to remove default groups `owner`, `shop-manager`, `guest`, and `customer`
1. See an error for each when you try
1. Ensure tests pass for update account group
1. Test updateAccountGroup to see that it still works

```graphql
# Get Shop ID
query shopId {
  primaryShopId
}

# Query groups
query groups {
  groups(shopId: "SHOP_ID") {
    nodes {
      _id
      name
      slug
    }
  }
}

# Remove queried group
mutation removeAccountGroup {
  removeAccountGroup(input: {
    groupId: "GROUP_ID",
    shopId: "SHOP_ID"
  }) {
    group {
      _id
      slug
      name
      permissions
    }
  }
}

# Update queried group
mutation updateAccountGroup {
  updateAccountGroup(input: {
    groupId: "GROUP_ID",
    shopId: "SHOP_ID"
    group: {
      name: "Customer Edited"
      slug: "customer"
      description: "Customers"
      permissions: ["permissions"] # This set all the permisisons (not append)
    }
  }) {
    group {
      _id
      slug
      name
      permissions
    }
  }
}
```
